src/ixdat/techniques/ms.py

```diff
@@ -1,7 +1,565 @@
 """Module for representation and analysis of MS measurements"""
 
-from ..measurements import Measurement
+import re
+import numpy as np
+import json  # FIXME: This is for MSCalibration.export, but shouldn't have to be here.
+
+from ..measurements import Measurement, Calibration
+from ..spectra import Spectrum
+from ..plotters.ms_plotter import MSPlotter, STANDARD_COLORS
+from ..exceptions import QuantificationError
+from ..constants import (
+    AVOGADROS_CONSTANT,
+    BOLTZMAN_CONSTANT,
+    STANDARD_TEMPERATURE,
+    STANDARD_PRESSURE,
+    DYNAMIC_VISCOSITIES,
+    MOLECULAR_DIAMETERS,
+    MOLAR_MASSES,
+)
+from ..data_series import ValueSeries
+from ..db import Saveable
 
 
 class MSMeasurement(Measurement):
     """Class implementing raw MS functionality"""
+
+    extra_column_attrs = {"ms_measurement": ("tspan_bg",)}
+    default_plotter = MSPlotter
+
+    def __init__(self, name, **kwargs):
+        tspan_bg = kwargs.pop("tspan_bg", None)
+        super().__init__(name, **kwargs)
+        self.tspan_bg = tspan_bg
+
+    @property
+    def ms_calibration(self):
+        ms_cal_list = []
+        tspan_bg = None
+        signal_bgs = {}
+        for cal in self.calibration_list:
+            ms_cal_list = ms_cal_list + getattr(cal, "ms_cal_list", [])
+            for mass, bg in getattr(cal, "signal_bgs", {}).items():
+                if mass not in signal_bgs:
+                    signal_bgs[mass] = bg
+            tspan_bg = tspan_bg or getattr(cal, "tspan_bg", None)
+        return MSCalibration(ms_cal_results=ms_cal_list, signal_bgs=signal_bgs)
+
+    @property
+    def signal_bgs(self):
+        return self.ms_calibration.signal_bgs
+
+    def set_bg(self, tspan_bg=None, mass_list=None):
+        """Set background values for mass_list to the average signal during tspan_bg."""
+        mass_list = mass_list or self.mass_list
+        tspan_bg = tspan_bg or self.tspan_bg
+        signal_bgs = {}
+        for mass in mass_list:
+            t, v = self.grab(mass, tspan_bg)
+            signal_bgs[mass] = np.mean(v)
+        self.add_calibration(MSCalibration(signal_bgs=signal_bgs))
+
+    def reset_bg(self, mass_list=None):
+        """Reset background values for the masses in mass_list"""
+        mass_list = mass_list or self.mass_list
+        for mass in mass_list:
+            if mass in self.signal_bgs:
+                del self.signal_bgs[mass]
+
+    def grab(
+        self,
+        item,
+        tspan=None,
+        tspan_bg=None,
+        include_endpoints=False,
+        removebackground=False,
+    ):
+        """Returns t, S where S is raw signal in [A] for a given signal name (ie mass)
+
+        Args:
+            item (str): Name of the signal.
```

Comment:
Ooh, or maybe it should be called `key` since that's what it's called in `Measurement.__getitem__()`. 
But it's called `item` in `Measurement.grab()`

I will leave it for now since it's a lot of refactoring with not much benefit (this will I think always be assigned as an arg, not a kwarg).

I've added a TODO in `Measurement.grab()`